Fix issue 44859

labkey-alan opened the pull request:
#### Rationale
Right now due to an implementation detail of HeaderWrapper anytime the URL is changed the page scrolls to the top. This is especially egregious when changing filters/sorts for grids, or when using the new ELN Home page. [This PR fixes that issue](https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=44859), and also reduces the need for duplicate code in our apps by slightly refactoring NavigationBar.

#### Related Pull Requests
* https://github.com/LabKey/biologics/pull/1195
* https://github.com/LabKey/sampleManagement/pull/875
* https://github.com/LabKey/inventory/pull/394

#### Changes
* Fix Issue 44859
  * HeaderWrapper no longer scrolls to the top of the page during update
  * Page component now scrolls to top of the page on mount
* Remove circular dependency between Page and NotFound
* Refactor NavigationBar to render HeaderWrapper and sticky class
  * This eliminates duplicated code between our apps
* Fix issue with URL binding pagination offsets when filters change

labkey-alan committed:
HeaderWrapper: Convert to FC, don't use jQuery, don't scroll on update

labkey-alan committed:
NavigationBar: render HeaderWrapper, sticky, and add children

This removes the need for a repeated set of boilerplate between all of our Apps

labkey-alan committed:
Page: refactor (see notes)

- Remove circular dependency between Page and NotFound
- Scroll to top of page during mount
- Don't use deprecated React methods
- Simplify some logic in render

labkey-alan committed:
Bump components, update release notes

labkey-alan committed:
Fix Page Not Found test

labkey-alan committed:
Update NavigationBar snapshots

cnathe reviewed (commented):
I was starting to CR this but noticed that I am still seeing the "scroll to top" behavior on grid filter change so I'll hold off. @labkey-alan is looking into that.

labkey-alan committed:
Don't wipe all rows when loading data after filters change

This fixes the issue with the page's scroll position changing
because the size of the page drastically changes

labkey-alan committed:
Bump components

labkey-alan committed:
QueryModel: Fix issue with offset not being reset when URL changes filters

labkey-alan committed:
Bump components

labkey-nicka reviewed (approved):
This is pretty swell. Tested it out locally and it things appear to be in working order. Things I tested:

1. URL changes no longer trigger page scroll.
1. Grid rows remain in place when grid inputs change.
1. Paging offset out of bounds when a filter is applied and page offset had been removed on URL.

labkey-alan committed:
Fix Docs URLs in QueryModel.ts

labkey-alan committed:
Bump components

labkey-alan committed:
Prep for release
